enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,974 @@
+
+## Version Skew Strategy
+
+N/A. This enhancement will only be supported for new installations and therefore has no version skew concerns.
+
+## Operational Aspects of API Extensions
+
+N/A.
+
+## Support Procedures
+
+### Ingress Not Functional After Installing using IngressController-role Subnets
+
+If the cluster installation with specified IngressController-role subnets was successful, but ingress is not working,
+check the AWS CCM logs of the leader CCM pod for errors in provisioning the load balancer:
+
+```bash
+oc logs -n openshift-cloud-controller-manager aws-cloud-controller-manager-86b68698cd-wfhgz
+```
+
+## Alternatives
+
+### Assume All Cluster-Role Subnets are IngressController-Role Subnets Alternative
+
+If the installer enforced that all cluster-role subnets are also IngressController-Role subnets, as suggested in
+the closed bug [OCPBUGS-17432](https://issues.redhat.com/browse/OCPBUGS-17432), then we could eliminate
+the need for user-facing install-config API updates and the deprecation of the `platform.aws.subnets` field.
+The installer would be updated to ensure that the values in `platform.aws.subnets` automatically propagates
+to the IngressConfig's `spec.loadBalancer.platform.aws.classicLoadBalancer.subnets` and
+`spec.loadBalancer.platform.aws.networkLoadBalancer.subnets` fields. This would make all cluster-role subnets
+the default IngressController subnets.
+
+While this would address the majority of customer use cases, this would invalidate the
+[Dedicated IngressController Subnets for Security](#dedicated-ingresscontroller-subnets-for-security) use case
+and could potentially be seen as a breaking change. Customers who previously relied on the selection of
+subnets without the `kubernetes.io/cluster/<cluster-id>` tag (introduced by [kubernetes/kubernetes#97431](https://github.com/kubernetes/kubernetes/pull/97431)),
+might find their future installs broken.
+
+### Assume All Subnets are Cluster-Role Subnets Alternative
+
+To simplify the install-config API and mitigate [No IngressController-Role Subnet in the Same AZ as Cluster-Role Subnets Risk](#no-ingresscontroller-role-subnet-in-the-same-az-as-cluster-role-subnets-risk),
+we could make the assumption that any provided subnets are cluster-role subnets. This would invalidate the user story
+[Dedicated IngressController Subnets for Security](#dedicated-ingresscontroller-subnets-for-security), but simplify the function of
+providing subnets via the install-config API.
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+    [...]
+    // SubnetsConfig specifies the subnets configuration for
+    // a cluster by specifying a list of subnets with their
+    // desired role. Leave unset to have the installer
+    // create subnets in a new VPC on your behalf.
+    //
+    // +optional
+    SubnetsConfig []SubnetConfig `json:"subnetsConfig,omitempty"`
```

Comment:
Did you bikeshed much on the name `SubnetsConfig`? We can't use `Subnets` as that is the field being deprecated... but I do think `Subnets` is better. It occurs to me that we could use `Subnets` with one level of misdirection by creating a VPC struct, so we could include the new fields as `installConfig.Platform.aws.vpc.subnets`. Thought I would throw that idea out there...